src/renderer/components/+helm-releases/release-details/release-details-content.tsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import "./release-details.scss";
+
+import React from "react";
+
+import { Link } from "react-router-dom";
+import { Drawer, DrawerItem, DrawerTitle } from "../../drawer";
+import { cssNames, stopPropagation } from "../../../utils";
+import { observer } from "mobx-react";
+import { withInjectables } from "@ogre-tools/injectable-react";
+import type { ConfigurationInput, MinimalResourceGroup, OnlyUserSuppliedValuesAreShownToggle, ReleaseDetailsModel } from "./release-details-model/release-details-model.injectable";
+import releaseDetailsModelInjectable from "./release-details-model/release-details-model.injectable";
+import { Button } from "../../button";
+import { kebabCase } from "lodash/fp";
+import { Badge } from "../../badge";
+import { SubTitle } from "../../layout/sub-title";
+import { Table, TableCell, TableHead, TableRow } from "../../table";
+import { ReactiveDuration } from "../../duration/reactive-duration";
+import { HelmReleaseMenu } from "../release-menu";
+import { Checkbox } from "../../checkbox";
+import { MonacoEditor } from "../../monaco-editor";
+import { Spinner } from "../../spinner";
+import type { TargetHelmRelease } from "./target-helm-release.injectable";
+
+interface ReleaseDetailsContentProps {
+  targetRelease: TargetHelmRelease;
+}
+
+interface Dependencies {
+  model: ReleaseDetailsModel;
+}
+
+const NonInjectedReleaseDetailsContent = observer(({ model }: Dependencies & ReleaseDetailsContentProps) => {
+  const isLoading = model.isLoading.get();
+
+  return (
+    <Drawer
+      className={cssNames("ReleaseDetails", model.activeTheme)}
+      usePortal={true}
+      open={true}
+      title={isLoading ? "" : model.release.getName()}
+      onClose={model.close}
+      testIdForClose="close-helm-release-detail"
+      toolbar={
+        !isLoading && (
+          <HelmReleaseMenu
+            release={model.release}
+            toolbar
+            hideDetails={model.close}
+          />
+        )
+      }
```

Comment:
Notice: Here the pain is that we have to set `title` and `toolbar` even though we don't have the data yet. Drawer needs to be refactored to place where we can set `title` and `toolbar` later.

It would look something like this:

```jsx
return (
  <Drawer open={true}>
    <DrawerHeader>
      <span>{model.release.getName()}</span>
     
      <HelmReleaseMenu toolbar release={model.release} />
    </DrawerHeader>

    <DrawerContent>
      Some content
    </DrawerContent>
  </Drawer>
)
```

In that case we could:

```jsx
return (
  <Drawer open={true}>
    <ReleaseDetailsContent />
  </Drawer>
);
```

And now `ReleaseDetailsContent` could have `async` dependency and `getPlaceholder` which returns spinner that looks good for drawer.

---

**Key thing is to start combining different components and expanding the scope of a single component by adding more props. That gives us flexibility, when latter only limits our ability to change our mind later.**

Reply:
Totally agree, let's do this.